backend/src/services/model.ts

```diff
@@ -457,3 +458,87 @@ export function isModelCardRevision(data: unknown): data is ModelCardRevisionInt
   }
   return true
 }
+
+export async function getCurrentUserPermissionsByModel(
+  user: UserInterface,
+  modelId: string,
+): Promise<EntryUserPermissions> {
+  const model = await getModelById(user, modelId)
+
+  const editEntryCardAuth = await authorisation.model(user, model, ModelAction.Write)
+  const viewEntrySettingsAuth = await authorisation.model(user, model, ModelAction.Update)
+  const createReleaseAuth = await authorisation.release(user, model, ReleaseAction.Create)
+  const editReleaseAuth = await authorisation.release(user, model, ReleaseAction.Update)
+  const deleteReleaseAuth = await authorisation.release(user, model, ReleaseAction.Delete)
+  const pushModelImageAuth = await authorisation.image(user, model, {
+    type: 'repository',
+    name: modelId,
+    actions: ['push'],
+  })
+  // Inferencing uses model authorisation
+  const createInferenceServiceAuth = await authorisation.model(user, model, ModelAction.Create)
+  const editInferenceServiceAuth = await authorisation.model(user, model, ModelAction.Update)
+  const exportMirroredModelAuth = await authorisation.model(user, model, ModelAction.Update)
+
+  return {
+    editEntryCard: editEntryCardAuth.success
```

Comment:
Is there a more succinct way of converting a standard auth response to user permission? I realise it's a bit fiddly because you need to ignore a property (id) and rename another property (success) when using the auth response. It might by nicer to define a utility function that takes in an auth response and returns a user permission object rather than have the same ternary statement repeated this many times